autoPyTorch/api/base_task.py

```diff
@@ -1,139 +1,740 @@
+                    'proba predictions: {}, predictions:{}'.format(
+                        prediction, pipeline.predict(X_, batch_size=batch_size)))
+                raise ValueError("For {}, prediction probability not within [0, 1]!".format(
+                    pipeline)
+                )
 
-class BaseTask():
+    if len(prediction.shape) < 1 or len(X_.shape) < 1 or \
+            X_.shape[0] < 1 or prediction.shape[0] != X_.shape[0]:
+        logger.warning(
+            "Prediction shape for model %s is %s while X_.shape is %s",
+            pipeline, str(prediction.shape), str(X_.shape)
+        )
+    return prediction
+
+
+class BaseTask:
     """
     Base class for the tasks that serve as API to the pipelines.
+    Args:
+        seed (int), (default=1): seed to be used for reproducibility.
+        n_jobs (int), (default=1): number of consecutive processes to spawn.
+        logging_config (Optional[Dict]): specifies configuration
+            for logging, if None, it is loaded from the logging.yaml
+        ensemble_size (int), (default=50): Number of models added to the ensemble built by
+            Ensemble selection from libraries of models.
+            Models are drawn with replacement.
+        ensemble_nbest (int), (default=50): only consider the ensemble_nbest
+            models to build the ensemble
+        max_models_on_disc (int), (default=50): maximum number of models saved to disc.
+            Also, controls the size of the ensemble as any additional models will be deleted.
+            Must be greater than or equal to 1.
+        temporary_directory (str): folder to store configuration output and log file
+        output_directory (str): folder to store predictions for optional test set
+        delete_tmp_folder_after_terminate (bool): determines whether to delete the temporary directory,
+            when finished
+        include_components (Optional[List[str]]): If None, all possible components are used.
+            Otherwise specifies set of components to use.
+        exclude_components (Optional[List[str]]): If None, all possible components are used.
+            Otherwise specifies set of components not to use. Incompatible with include
+            components
     """
 
     def __init__(
-        self
-    ):
-        self.pipeline = self.build_pipeline()
+            self,
+            seed: int = 1,
+            n_jobs: int = 1,
+            logging_config: Optional[Dict] = None,
+            ensemble_size: int = 50,
+            ensemble_nbest: int = 50,
+            max_models_on_disc: int = 50,
+            temporary_directory: Optional[str] = None,
+            output_directory: Optional[str] = None,
+            delete_tmp_folder_after_terminate: bool = True,
+            delete_output_folder_after_terminate: bool = True,
+            include_components: Optional[Dict] = None,
+            exclude_components: Optional[Dict] = None,
+            backend: Optional[Backend] = None,
```

Comment:
Do you think there is a way to reduce the number of arguments here and in `fit`? E.g. could we pack all pipeline options to the default pipeline config (isn't seed for instance a pipeline option?). My worry is that we have a lot flying around with the arguments from `fit`, `init` and `self.default_pipeline_options` and that things dont get updates/used properly

Reply:
I agree with seed being a pipeline option but it's also used in smac and ensemble. And the other options all specify attributes which are used for the whole experiment and not in the pipeline config. All I can think of is maybe removing budget and budget type, but for that I feel, its a good option to allow just for search or refit a different budget.

Reply:
Okay then let's keep it as is